### ISSUE TITLE
NPE in PerformanceWatcher

### ISSUE THREAD

**Issue by tpasternak** [bazelbuild/intellij#6577]
### Description of the bug:

```
RuntimeException while executing runnable AsyncTransformFuture@7a1b37c1[status=PENDING, info=[inputFuture=[TrustedListenableFutureTask@c11a899[status=SUCCESS, result=[[Ljava.io.File;@1ca49641]]], function=[com.google.idea.blaze.base.sync.projectstructure.DirectoryStructure$$Lambda/0x00000008031132e0@31506672]]] with executor com.google.common.util.concurrent.MoreExecutors$5@1256eaa6

java.lang.NullPointerException: Cannot invoke "kotlin.coroutines.CoroutineContext.get(kotlin.coroutines.CoroutineContext$Key)" because "context" is null
	at com.intellij.diagnostic.CoroutineDumperKt.toRepresentation(coroutineDumper.kt:281)
	at com.intellij.diagnostic.CoroutineDumperKt.toRepresentation(coroutineDumper.kt:292)
	at com.intellij.diagnostic.CoroutineDumperKt.toRepresentation(coroutineDumper.kt:292)
	at com.intellij.diagnostic.CoroutineDumperKt.toRepresentation(coroutineDumper.kt:292)
	at com.intellij.diagnostic.CoroutineDumperKt.toRepresentation(coroutineDumper.kt:292)
	at com.intellij.diagnostic.CoroutineDumperKt.toRepresentation(coroutineDumper.kt:292)
	at com.intellij.diagnostic.CoroutineDumperKt.toRepresentation(coroutineDumper.kt:292)
	at com.intellij.diagnostic.CoroutineDumperKt.toRepresentation(coroutineDumper.kt:292)
	at com.intellij.diagnostic.CoroutineDumperKt.toRepresentation(coroutineDumper.kt:292)
	at com.intellij.diagnostic.CoroutineDumperKt.toRepresentation(coroutineDumper.kt:292)
	at com.intellij.diagnostic.CoroutineDumperKt.dumpCoroutines(coroutineDumper.kt:123)
	at com.intellij.diagnostic.CoroutineDumperKt.dumpCoroutines(coroutineDumper.kt:58)
	at com.intellij.diagnostic.ThreadDumper.getThreadDumpInfo(ThreadDumper.java:68)
	at com.intellij.diagnostic.PerformanceWatcherImpl.doDumpThreads(PerformanceWatcherImpl.kt:261)
	at com.intellij.diagnostic.PerformanceWatcherImpl.access$doDumpThreads(PerformanceWatcherImpl.kt:70)
	at com.intellij.diagnostic.PerformanceWatcherImpl$asyncInit$4$1.invoke(PerformanceWatcherImpl.kt:155)
	at com.intellij.diagnostic.PerformanceWatcherImpl$asyncInit$4$1.invoke(PerformanceWatcherImpl.kt:150)
	at com.intellij.diagnostic.PerformanceWatcherImpl$asyncInit$4.invokeSuspend$lambda$0(PerformanceWatcherImpl.kt:150)
	at com.intellij.util.concurrency.AppScheduledExecutorService$MyThreadFactory.newThread(AppScheduledExecutorService.java:61)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.<init>(ThreadPoolExecutor.java:637)
	at java.base/java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:928)
	at java.base/java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1375)
	at com.intellij.util.concurrency.AppScheduledExecutorService$BackendThreadPoolExecutor.execute(AppScheduledExecutorService.java:148)
	at com.intellij.util.concurrency.BoundedTaskExecutor.wrapAndExecute(BoundedTaskExecutor.java:238)
	at com.intellij.util.concurrency.BoundedTaskExecutor.execute(BoundedTaskExecutor.java:162)
	at com.google.common.util.concurrent.MoreExecutors$ListeningDecorator.execute(MoreExecutors.java:640)
	at com.google.common.util.concurrent.MoreExecutors$5.execute(MoreExecutors.java:1167)
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1298)
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:1059)
	at com.google.common.util.concurrent.AbstractFuture.set(AbstractFuture.java:784)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.afterRanInterruptiblySuccess(TrustedListenableFutureTask.java:136)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:89)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:27)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:249)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:30)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.executeFirstTaskAndHelpQueue(BoundedTaskExecutor.java:227)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:218)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:212)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:735)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:732)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:732)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

### Which Intellij IDE are you using? Please provide the specific version.

2024.2 Beta 1

**Comment by tpasternak** [comment]
https://youtrack.jetbrains.com/issue/IJPL-158517/NPE-in-PerformanceWatcher

**Comment by tpasternak** [comment]
Fixed in IJ 2024.2
